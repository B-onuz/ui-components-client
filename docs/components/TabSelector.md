# TabSelector (Component)

Componente de seletor de abas.

## Como utilizar

```js
import TabSelector from 'ui-components/lib/components/TabSelector'
```

```jsx
import React, { useState } from 'react'
import TabSelector from './TabSelector'
import Fieldset from '../Fieldset'
import TextField from '../TextField'
import { Box, Flex } from 'reflexbox'

export default {
  title: 'TabSelector',
  component: TabSelector,
}

export const TabSelectorExample = () => {
  const [contentA, setContentA] = useState(true)
  const [contentB, setContentB] = useState(false)

  const handleTab = () => {
    setContentA(!contentA)
    setContentB(!contentB)
  }

  return (
    <Flex flexDirection={'column'} alignItems={'center'}>
      <TabSelector
        color={'primary'}
        optionA={'Pessoa física'}
        optionB={'Pessoa jurídica'}
        activeA={contentA}
        activeB={contentB}
        handleClick={handleTab}
      />
      {contentA && (
        <Fieldset m={4} legend={'Informações pessoais'}>
          <Flex flexWrap={'wrap'} m={-2}>
            <Box width={[1, 1 / 2, 1 / 3]} m={2}>
              <TextField id="name" name={'name'} required label={'Nome'} placeholder={'Nome'} />
            </Box>
            <Box width={[1, 1 / 2, 1 / 3]} m={2}>
              <TextField id="lastname" name={'lastname'} label={'Sobrenome'} placeholder={'Sobrenome'} />
            </Box>
            <Box width={[1, 1 / 2, 1 / 3]} m={2}>
              <TextField id="fullname" name={'fullname'} label={'Exibir Como:'} placeholder={'Nome + Sobrenome'} />
            </Box>
          </Flex>
        </Fieldset>
      )}
      {contentB && (
        <Fieldset m={4} bordered p={'1em'} legend={'Dados de contato (bordered)'}>
          <Flex flexWrap={'wrap'} m={-2}>
            <Box width={[1, 1 / 2, 1 / 3]} m={2}>
              <TextField id="email" type="email" name={'email'} label={'Email'} placeholder={'Email'} />
            </Box>
            <Box width={[1, 1 / 2, 1 / 3]} m={2}>
              <TextField id="telefone" type="phone" name={'fullname'} label={'Telefone'} placeholder={'Telefone'} />
            </Box>
          </Flex>
        </Fieldset>
      )}
    </Flex>
  )
}
```

## Styled-system props

- space https://styled-system.com/table#space
