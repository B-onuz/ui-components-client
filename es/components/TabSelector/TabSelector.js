"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _styledSystem = require("styled-system");

var _Button = _interopRequireDefault(require("../Button"));

var _this = void 0,
    _jsxFileName = "/Users/master/projects/bonuz/ui-components/src/components/TabSelector/TabSelector.js";

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n  display: inline-block;\n  border-radius: 25px;\n  overflow: hidden;\n  border: 2px solid ", ";\n  @media only screen and (max-width: 500px) {\n    display: flex;\n    flex-direction: column;\n    border-radius: 12px;\n  }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  ", "\n  @media only screen and (max-width: 500px) {\n    min-width: 180px;\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

var StyledTabSelector = _styledComponents["default"].div(_templateObject(), _styledSystem.color);

var ButtonsWrapper = _styledComponents["default"].div(_templateObject2(), function (_ref) {
  var theme = _ref.theme,
      color = _ref.color;
  return "" + theme.colors[color];
});

var TabSelector = function TabSelector(_ref2) {
  var activeA = _ref2.activeA,
      activeB = _ref2.activeB,
      optionA = _ref2.optionA,
      optionB = _ref2.optionB,
      color = _ref2.color,
      handleClick = _ref2.handleClick,
      rest = _objectWithoutPropertiesLoose(_ref2, ["activeA", "activeB", "optionA", "optionB", "color", "handleClick"]);

  return /*#__PURE__*/_react["default"].createElement(StyledTabSelector, _extends({}, rest, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 27,
      columnNumber: 5
    }
  }), /*#__PURE__*/_react["default"].createElement(ButtonsWrapper, {
    color: color,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 28,
      columnNumber: 7
    }
  }, /*#__PURE__*/_react["default"].createElement(_Button["default"], {
    linkButton: !activeA,
    color: color,
    py: 1,
    style: {
      borderRadius: 0
    },
    onClick: handleClick,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 29,
      columnNumber: 9
    }
  }, optionA), /*#__PURE__*/_react["default"].createElement(_Button["default"], {
    linkButton: !activeB,
    color: color,
    py: 1,
    style: {
      borderRadius: 0
    },
    onClick: handleClick,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 32,
      columnNumber: 9
    }
  }, optionB)));
};

var _default = TabSelector;
exports["default"] = _default;