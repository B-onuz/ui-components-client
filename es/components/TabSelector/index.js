"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _TabSelector = _interopRequireDefault(require("./TabSelector"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var _default = _TabSelector["default"];
exports["default"] = _default;