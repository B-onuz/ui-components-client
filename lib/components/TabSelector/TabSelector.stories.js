'use strict'

function _typeof(obj) {
  '@babel/helpers - typeof'
  if (typeof Symbol === 'function' && typeof Symbol.iterator === 'symbol') {
    _typeof = function _typeof(obj) {
      return typeof obj
    }
  } else {
    _typeof = function _typeof(obj) {
      return obj && typeof Symbol === 'function' && obj.constructor === Symbol && obj !== Symbol.prototype ? 'symbol' : typeof obj
    }
  }
  return _typeof(obj)
}

exports.__esModule = true
exports.TabSelectorExample = exports['default'] = void 0

var _react = _interopRequireWildcard(require('react'))

var _TabSelector = _interopRequireDefault(require('./TabSelector'))

var _Fieldset = _interopRequireDefault(require('../Fieldset'))

var _TextField = _interopRequireDefault(require('../TextField'))

var _reflexbox = require('reflexbox')

var _this = void 0,
  _jsxFileName = '/home/junior/Fontes/ui-components-client/src/components/TabSelector/TabSelector.stories.js'

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj }
}

function _getRequireWildcardCache() {
  if (typeof WeakMap !== 'function') return null
  var cache = new WeakMap()
  _getRequireWildcardCache = function _getRequireWildcardCache() {
    return cache
  }
  return cache
}

function _interopRequireWildcard(obj) {
  if (obj && obj.__esModule) {
    return obj
  }
  if (obj === null || (_typeof(obj) !== 'object' && typeof obj !== 'function')) {
    return { default: obj }
  }
  var cache = _getRequireWildcardCache()
  if (cache && cache.has(obj)) {
    return cache.get(obj)
  }
  var newObj = {}
  var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor
  for (var key in obj) {
    if (Object.prototype.hasOwnProperty.call(obj, key)) {
      var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null
      if (desc && (desc.get || desc.set)) {
        Object.defineProperty(newObj, key, desc)
      } else {
        newObj[key] = obj[key]
      }
    }
  }
  newObj['default'] = obj
  if (cache) {
    cache.set(obj, newObj)
  }
  return newObj
}

var _default = {
  title: 'TabSelector',
  component: _TabSelector['default'],
}
exports['default'] = _default

var TabSelectorExample = function TabSelectorExample() {
  var _useState = (0, _react.useState)(true),
    contentA = _useState[0],
    setContentA = _useState[1]

  var _useState2 = (0, _react.useState)(false),
    contentB = _useState2[0],
    setContentB = _useState2[1]

  var handleTab = function handleTab() {
    setContentA(!contentA)
    setContentB(!contentB)
  }

  return /*#__PURE__*/ _react['default'].createElement(
    _reflexbox.Flex,
    {
      flexDirection: 'column',
      alignItems: 'center',
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 22,
        columnNumber: 5,
      },
    },
    /*#__PURE__*/ _react['default'].createElement(_TabSelector['default'], {
      color: 'primary',
      optionA: 'Pessoa física',
      optionB: 'Pessoa jurídica',
      activeA: contentA,
      activeB: contentB,
      handleClick: handleTab,
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 23,
        columnNumber: 7,
      },
    }),
    contentA &&
      /*#__PURE__*/ _react['default'].createElement(
        _Fieldset['default'],
        {
          m: 4,
          legend: 'Informações pessoais',
          __self: _this,
          __source: {
            fileName: _jsxFileName,
            lineNumber: 32,
            columnNumber: 9,
          },
        },
        /*#__PURE__*/ _react['default'].createElement(
          _reflexbox.Flex,
          {
            flexWrap: 'wrap',
            m: -2,
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 33,
              columnNumber: 11,
            },
          },
          /*#__PURE__*/ _react['default'].createElement(
            _reflexbox.Box,
            {
              width: [1, 1 / 2, 1 / 3],
              m: 2,
              __self: _this,
              __source: {
                fileName: _jsxFileName,
                lineNumber: 34,
                columnNumber: 13,
              },
            },
            /*#__PURE__*/ _react['default'].createElement(_TextField['default'], {
              id: 'name',
              name: 'name',
              required: true,
              label: 'Nome',
              placeholder: 'Nome',
              __self: _this,
              __source: {
                fileName: _jsxFileName,
                lineNumber: 35,
                columnNumber: 15,
              },
            })
          ),
          /*#__PURE__*/ _react['default'].createElement(
            _reflexbox.Box,
            {
              width: [1, 1 / 2, 1 / 3],
              m: 2,
              __self: _this,
              __source: {
                fileName: _jsxFileName,
                lineNumber: 37,
                columnNumber: 13,
              },
            },
            /*#__PURE__*/ _react['default'].createElement(_TextField['default'], {
              id: 'lastname',
              name: 'lastname',
              label: 'Sobrenome',
              placeholder: 'Sobrenome',
              __self: _this,
              __source: {
                fileName: _jsxFileName,
                lineNumber: 38,
                columnNumber: 15,
              },
            })
          ),
          /*#__PURE__*/ _react['default'].createElement(
            _reflexbox.Box,
            {
              width: [1, 1 / 2, 1 / 3],
              m: 2,
              __self: _this,
              __source: {
                fileName: _jsxFileName,
                lineNumber: 40,
                columnNumber: 13,
              },
            },
            /*#__PURE__*/ _react['default'].createElement(_TextField['default'], {
              id: 'fullname',
              name: 'fullname',
              label: 'Exibir Como:',
              placeholder: 'Nome + Sobrenome',
              __self: _this,
              __source: {
                fileName: _jsxFileName,
                lineNumber: 41,
                columnNumber: 15,
              },
            })
          )
        )
      ),
    contentB &&
      /*#__PURE__*/ _react['default'].createElement(
        _Fieldset['default'],
        {
          m: 4,
          bordered: true,
          p: '1em',
          legend: 'Dados de contato (bordered)',
          __self: _this,
          __source: {
            fileName: _jsxFileName,
            lineNumber: 47,
            columnNumber: 9,
          },
        },
        /*#__PURE__*/ _react['default'].createElement(
          _reflexbox.Flex,
          {
            flexWrap: 'wrap',
            m: -2,
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 48,
              columnNumber: 11,
            },
          },
          /*#__PURE__*/ _react['default'].createElement(
            _reflexbox.Box,
            {
              width: [1, 1 / 2, 1 / 3],
              m: 2,
              __self: _this,
              __source: {
                fileName: _jsxFileName,
                lineNumber: 49,
                columnNumber: 13,
              },
            },
            /*#__PURE__*/ _react['default'].createElement(_TextField['default'], {
              id: 'email',
              type: 'email',
              name: 'email',
              label: 'Email',
              placeholder: 'Email',
              __self: _this,
              __source: {
                fileName: _jsxFileName,
                lineNumber: 50,
                columnNumber: 15,
              },
            })
          ),
          /*#__PURE__*/ _react['default'].createElement(
            _reflexbox.Box,
            {
              width: [1, 1 / 2, 1 / 3],
              m: 2,
              __self: _this,
              __source: {
                fileName: _jsxFileName,
                lineNumber: 52,
                columnNumber: 13,
              },
            },
            /*#__PURE__*/ _react['default'].createElement(_TextField['default'], {
              id: 'telefone',
              type: 'phone',
              name: 'fullname',
              label: 'Telefone',
              placeholder: 'Telefone',
              __self: _this,
              __source: {
                fileName: _jsxFileName,
                lineNumber: 53,
                columnNumber: 15,
              },
            })
          )
        )
      )
  )
}

exports.TabSelectorExample = TabSelectorExample
