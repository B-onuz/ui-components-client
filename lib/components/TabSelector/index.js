"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _TabSelector = _interopRequireDefault(require("./TabSelector"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var _default = _TabSelector["default"];
exports["default"] = _default;
module.exports = exports.default;