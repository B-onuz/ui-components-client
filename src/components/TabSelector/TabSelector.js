import React, { useState } from 'react'
import styled from 'styled-components'
import { color } from 'styled-system'
import Button from '../Button'

const StyledTabSelector = styled.div`
  ${color}
  @media only screen and (max-width: 500px) {
    min-width: 180px;
  }
`

const ButtonsWrapper = styled.div`
  display: inline-block;
  border-radius: 25px;
  overflow: hidden;
  border: 2px solid ${({ theme, color }) => `${theme.colors[color]}`};
  @media only screen and (max-width: 500px) {
    display: flex;
    flex-direction: column;
    border-radius: 12px;
  }
`

const TabSelector = ({ activeA, activeB, optionA, optionB, color, handleClick, ...rest }) => {
  return (
    <StyledTabSelector {...rest}>
      <ButtonsWrapper color={color}>
        <Button linkButton={!activeA} color={color} py={1} style={{ borderRadius: 0 }} onClick={handleClick}>
          {optionA}
        </Button>
        <Button linkButton={!activeB} color={color} py={1} style={{ borderRadius: 0 }} onClick={handleClick}>
          {optionB}
        </Button>
      </ButtonsWrapper>
    </StyledTabSelector>
  )
}

export default TabSelector
