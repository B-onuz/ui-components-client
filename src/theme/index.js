const colors = {
  red: '#db2828',
  lightRed: '#ef4d5a',
  mediumRed: '#FF4F5D',
  darkRed: '#9c2543',
  fullDarkRed: '#550e33',
  orange: '#f2711c',
  yellow: '#fbbd08',
  olive: '#b5cc18',
  green: '#21BA45',
  teal: '#00b5ad',
  blue: '#2185d0',
  facebookBlue: '#3B5998',
  blueLink: '#4285F4',
  violet: '#6435c9',
  purple: '#a333c8',
  pink: '#e03997',
  brown: '#a5673f',
  grey: '#767676',
  textGrey: '#8D8D8D',
  lightGrey: '#6D6D6D',
  mediumGrey: '#838383',
  darkGrey: '#9A9A9A',
  boldGray: '#55555A',
  black: '#1b1c1d',
  white: '#fdfdfd',
  warning: '#7A4D05',
  success: '#1E561F',
  info: '#0E566C',
  danger: '#973937',
  cyan: '#1CB5AD',
  disabledGrey: '#ffffff1f',
  disabledGreyText: '#ffffff4d',
}

colors.primary = colors.lightRed
colors.secondary = colors.fullDarkRed
colors.default = '#e0e1e2'

const breakpoints = ['40em', '52em', '64em', '80em']

breakpoints.sm = breakpoints[0]
breakpoints.md = breakpoints[1]
breakpoints.lg = breakpoints[2]
breakpoints.xl = breakpoints[3]

const fontSizes = [12, 14, 16, 20, 24, 32]

export default {
  fontFamily: `'Lato', sans-serif`,
  colors,
  fontSizes,
  breakpoints,
}
