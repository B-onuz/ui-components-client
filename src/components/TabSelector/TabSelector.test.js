import React from 'react'
import TabSelector from './TabSelector.js'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import theme from '../../theme'
import 'jest-styled-components'

test('Should render TabSelector', () => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <TabSelector />
    </ThemeProvider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
