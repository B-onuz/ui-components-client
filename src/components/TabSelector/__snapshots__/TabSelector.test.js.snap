// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render TabSelector 1`] = `
.c2 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font-family: inherit;
  border-radius: 4px;
  outline: none;
  cursor: pointer;
  padding: .5rem 1rem;
  border-style: solid;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  border-color: undefined !important;
  color: #000;
  background-color: #e0e1e2;
  border-color: #e0e1e2;
  font-size: 16px;
  padding-top: 4px;
  padding-bottom: 4px;
  box-sizing: border-box;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #e0e1e2;
  border: none;
  background-color: transparent !important;
  box-shadow: none !important;
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2:focus {
  box-shadow: 0 0 8px #e0e1e2;
}

.c2:active:not(:disabled) {
  box-shadow: inset 0px 0px 5px #c1c1c1;
}

.c2:active,
.c2:hover {
  background-color: hsl(210,3.7%,79.4%);
}

.c2:disabled {
  background-color: hsl(210,2%,97.1%);
}

.c2:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  background-color: transparent;
}

.c1 {
  display: inline-block;
  border-radius: 25px;
  overflow: hidden;
  border: 2px solid undefined;
}

@media only screen and (max-width:500px) {
  .c0 {
    min-width: 180px;
  }
}

@media only screen and (max-width:500px) {
  .c1 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    border-radius: 12px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <button
      className="c2"
      color="default"
      disabled={false}
      size="medium"
      style={
        Object {
          "borderRadius": 0,
        }
      }
      type="button"
    />
    <button
      className="c2"
      color="default"
      disabled={false}
      size="medium"
      style={
        Object {
          "borderRadius": 0,
        }
      }
      type="button"
    />
  </div>
</div>
`;
